### PR TITLE
Fixes #358 Allow user to specify branch to generate docs

### LINKED ISSUE #358: Allow the user to specify a branch to generate docs
### This is a
- [ ] Bug report
- [x] Feature request

### Current behaviour

The documentation for a registered repository is generated at every commit made to the repository. This process happens in commits made to any of the repository's branches except for branches that do not have a `.yaydoc.yml` file and the `gh-pages` branch.
### Expected behaviour

It would be nice if the user can limit the doc generation process to a specific branch or a couple of branches

### PR DESCRIPTION
**Pull request by imujjwal96**
## Description

Allow users to specify branches from their registered repositories settings at Yaydoc. User may also delete some of the specified branches. 
If no branches are specified, documentation will be generated from all branches containing `.yaydoc.yml` file, except the gh-pages branch
## Related Issue

Fixes #358 
## Motivation and Context

The documentation for a registered repository is generated at every commit made to the repository. This process happens in commits made to any of the repository's branches except for branches that do not have a .yaydoc.yml file and the gh-pages branch. It would be nice if the user can limit the doc generation process to a specific branch or a couple of branches
## How Has This Been Tested?

Test at https://yaydoc7.herokuapp.com

## Types of changes

- [ ] Bug fix 
- [x] New feature

## Checklist:

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] There is a corresponding issue for this pull request.
- [x] Mentioned the Issue number in the pull request commit message `Fixes #<number> commit message`
- [x] There is only one commit per issue.

### CODE REVIEW

**Comment by imujjwal96** [comment]
@pri22296 @sch00lb0y Please review.

File: public/scripts/repository/settings.js [hunk 1]
```diff
@@ -1,10 +1,9 @@
 $(function() {
   const predefinedMessages = {
-    'mail_disabled_successful': "Mail service disabled for this repository",
```
Comment (pri22296) [inline public/scripts/repository/settings.js]: Are these messages no longer needed?

Comment (imujjwal96) [reply public/scripts/repository/settings.js]: No we are not showing them now. We have added a toggle for these events and we show a check mark beside the element when the change is successful

**Comment by pri22296** [comment]
@imujjwal96 Shows null after adding and deleting branches

screenshot 218

**Comment by imujjwal96** [comment]
@pri22296 I am not able to reproduce this error

**Comment by pri22296** [comment]
@imujjwal96 Let me test this more and i'll get back to you If I can reproduce it.

**Comment by pri22296** [comment]
@imujjwal96 I'm not able to reproduce that error. It wold be great though if you could take a look at the code and just ensure that everything is fine.

Also submit button gets hidden when there are >=3 branches.
screenshot 219

**Comment by imujjwal96** [comment]
> Also submit button gets hidden when there are >=3 branches.
does it create any problem? I thought users may not submit the form without closing the select tag

**Comment by pri22296** [comment]
The UI feels awkward there. consider showing the branches with a scroll bar and having the submit button slightly below.

**Comment by imujjwal96** [comment]
@pri22296 Updated it.

File: backend/github.js [hunk 2]
```diff
@@ -236,4 +236,30 @@ exports.getUserDetail = function (userName, callback) {
...
+ */
+exports.getRepositoryBranches = function (name, callback) {
+  request.get({
+    url: 'https://api.github.com/repos/' + name + '/branches',
+    headers: {
+      'User-Agent': 'Yaydoc',
+    },
+  }, function (error, response, body) {
+    if (error) {
+      callback(error);
```
Comment (sch00lb0y) [inline backend/github.js]: pass `null` in the second parameter. it look consistent

**Comment by imujjwal96** [comment]
@pri22296 @sch00lb0y Updated

**Pull request merged**